app/components/syntax_highlight/renderer.tsx

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback} from 'react';
+import {FlatList, ScrollView, StyleSheet, Text} from 'react-native';
+import {createStyleObject} from 'react-syntax-highlighter/create-element';
+
+import {changeOpacity} from '@utils/theme';
+
+type CreateChildren = {
+    digits: number;
+    stylesheet: any;
+    fontFamily: string;
+    fontSize?: number;
+    selectable?: boolean;
+}
+
+type CreateNativeElement = CreateChildren & {
+    node: any;
+    key: string;
+    defaultColor: string;
+}
+
+type Props = CreateChildren & Pick<CreateNativeElement, 'defaultColor'> & {
+    rows: any[];
+}
+
+const styles = StyleSheet.create({
+    wrapped: {
+        flexWrap: 'nowrap',
+    },
+});
+
+function createChildren({stylesheet, fontSize = 12, fontFamily, selectable, digits}: CreateChildren) {
+    let childrenCount = 0;
+    return (children: any[], defaultColor: string) => {
+        childrenCount += 1;
+        return children.map((child, i) => {
+            return createNativeElement({
+                node: child,
+                stylesheet,
+                key: `code-segment-${childrenCount}-${i}`,
+                defaultColor,
+                fontSize,
+                fontFamily,
+                selectable,
+                digits,
+            });
+        });
+    };
+}
+
+function createNativeElement({node, stylesheet, key, defaultColor, fontFamily, fontSize = 12, selectable, digits}: CreateNativeElement) {
+    const {properties, type, tagName: TagName, value} = node;
+    const startingStyle = {fontFamily, fontSize, height: fontSize + 7};
+    if (properties?.key?.startsWith('line-number')) {
+        let valueString = `${node.children[0].value}. `;
+        for (let i = valueString.length; i < digits + 2; i++) {
+            valueString += ' ';
+        }
+
+        return (
+            <Text
+                key={key}
+                style={{color: changeOpacity(defaultColor, 0.75), paddingRight: 5, ...startingStyle}}
```

Comment:
New object on every render. Since it does not have almost anything as a child, may be OK to leave like this.

Reply:
I don't see the lines being re-rendered at this point unless parent re-renders with a code block change